.circleci/config.yml

```diff
@@ -593,54 +558,54 @@ jobs:
         default: []
     environment:
       WORKBENCH_ENV: << parameters.env_name >>
-      TEST_MODE: << parameters.test_mode >>
     parallelism: << parameters.parallel_num >>
     steps:
-      - checkout-code
+      - attach_workspace:
+          at: .
       - steps: << parameters.optional_steps >>
-      - halt-test-check:
-          dir_names: "api ui e2e"
       - run:
-          name: Update test user environment variables
-          command: |
-            case << parameters.env_name >> in
-              "staging")
-                echo 'export STAGING_COLLABORATOR=$PUPPETEER_COLLABORATOR_STAGING' >> $BASH_ENV
-                echo 'export USER_NAME=$PUPPETEER_USER_STAGING@staging.fake-research-aou.org' >> $BASH_ENV
-                ;;
-              "test")
-                echo 'export TEST_COLLABORATOR=$PUPPETEER_COLLABORATOR_TEST' >> $BASH_ENV
-                echo 'export USER_NAME=$PUPPETEER_USER_TEST@fake-research-aou.org' >> $BASH_ENV
-                ;;
-              "local")
-                echo 'export DEV_COLLABORATOR=$PUPPETEER_COLLABORATOR_LOCAL' >> $BASH_ENV
-                echo 'export USER_NAME=$PUPPETEER_USER_LOCAL@fake-research-aou.org' >> $BASH_ENV
-                ;;
-            esac
-            echo 'export PASSWORD=$PUPPETEER_USER_PASSWORD' >> $BASH_ENV
-            source $BASH_ENV
+          name: "Update environment variables for e2e test"
+          command: bash .circleci/e2e-test-users.sh << parameters.env_name >>
       - when:
           condition:
             equal: [ "local", << parameters.env_name >> ]
           steps:
-            # Local UI server is not used if WORKBENCH_ENV reset to "test" value in next step
-            - launch-local-ui
             - run:
-                name: See if WORKBENCH_ENV environment variable should override from "local" to "test".
-                command: |
-                  if [ ${CIRCLE_BRANCH} != "" ] && [ ${CIRCLE_BRANCH} != "master" ]; then
-                    # Count total number of files changed.
-                    CHANGED_COUNT=$(git diff --name-only $(git merge-base origin/master ${CIRCLE_BRANCH}) | wc -l | xargs)
-                    E2E_CHANGED_COUNT=$(git diff --name-only $(git merge-base origin/master ${CIRCLE_BRANCH}) -- e2e | wc -l | xargs)
-                    # If e2e tests are the only files that have changed, run e2e tests against the "test" environment.
-                    if [ $E2E_CHANGED_COUNT -gt 0 ] && [ $E2E_CHANGED_COUNT -eq $CHANGED_COUNT ]; then
-                        echo 'export WORKBENCH_ENV=test' >> $BASH_ENV
-                        echo 'export USER_NAME=${PUPPETEER_USER_TEST}@fake-research-aou.org' >> $BASH_ENV
-                        source $BASH_ENV
-                        echo "Reset WORKBENCH_ENV=test"
-                    fi
-                  fi
-      - run-puppeteer-test
+                name: Change WORKBENCH_ENV to "test" if needed
+                command: bash .circleci/e2e-update-env.sh
```

Comment:
suggest: `e2e-choose-env.sh` or `e2e-pick-env.sh`

Reply:
Renamed to `choose-e2e-env.sh` (verb first)